In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import os


%matplotlib inline

# Limit rows disp# Limit rows displayed in notebook
pd.set_option('display.max_rows', 20)
pd.set_option('display.precision', 2)

In [2]:
reviews = pd.read_pickle('review/reviews_with_topics.pkl')
reviews.head()

business_id               review_id date.year date.month  \
0  E7nk0lC317pGxMX56gvaVQ  onOLEBffcpWS_jcaBLCqdw      2008          6   
1  duHUQFn7K-Ybs1rPI4AOEQ  23z4mKkaMCXGFw4cm6xHxA      2012          4   
2  _K5cfCM7-zQZ1hXdbsc2fQ  dqrwXpcYdFvNYL5z1AeuZw      2013         12   
3  OTq55xUFMOz13DAi9D0_rw  uiJbTytfC1zsSOQ4aA3AWg      2014         12   
4  Kfbs3K37ZczrOQnoxgzNkw  II7cxU1cbOxVqzYY4ueflQ      2012          1   

  date.day stars                                               text  \
0       17     1  This place is a huge waste of the space they m...   
1       15     3  I used to frequent Z-Tejas more frequently for...   
2        1     3  Meeting some friends for dinner, they live in ...   
3       23     4                        Good food, and good service   
4       25     4  Came here today to get lunch to go. They were ...   

  review_length votes.cool votes.funny votes.useful                 user_id  \
0          1239          3           4            2  eBFm-lABQiKpaUcPDfYOgA   
1           936          1           1            2  y5_vrqIylhXMrejM_-x_vA   
2          1119          0           0            0  8si1MT348c-rNZYDWCJvzQ   
3            27          0           0            1  YfWmj8wWbuiCbnC6D8gvpw   
4           661          1           0            2  kffxYQVQsEpF1DVlQlcGCg   

   topic_id_50  topic_prob_50  \
0           39            0.6   
1           43            0.7   
2           39            0.6   
3           48            0.9   
4            5            0.9   

                                        word_dist_50  
0  [(0.00977830773598, place), (0.00650128695031,...  
1  [(0.0090865194881, good), (0.00526586990462, g...  
2  [(0.00977830773598, place), (0.00650128695031,...  
3  [(0.0146616328294, good), (0.0142812265826, fo...  
4  [(0.0063096357766, food), (0.00550394162605, g...

In [3]:
reviews.describe()

topic_id_50  topic_prob_50
count      10000.0        10000.0
mean          27.1            0.5
std           14.7            0.2
min            0.0            0.0
25%           15.0            0.3
50%           28.0            0.5
75%           39.0            0.6
max           49.0            1.0

In [4]:
restaurants = pd.read_pickle('restaurant/restaurants5k.pkl')
print restaurants.describe()
restaurants.head()

         open  review_count   stars  cat.Active Life  cat.Adult Entertainment  \
count  3358.0        3358.0  3358.0          3.4e+03                     3358   
mean      0.9          85.5     3.6          2.4e-03                        0   
std       0.3          72.5     0.6          4.9e-02                        0   
min       0.0           3.0     1.5          0.0e+00                        0   
25%       1.0          34.0     3.5          0.0e+00                        0   
50%       1.0          62.0     3.5          0.0e+00                        0   
75%       1.0         113.0     4.0          0.0e+00                        0   
max       1.0         384.0     5.0          1.0e+00                        0   

       cat.Afghan  cat.African  cat.Airports  cat.American (New)  \
count     3.4e+03      3.4e+03          3358              3358.0   
mean      8.9e-04      6.0e-04             0                 0.1   
std       3.0e-02      2.4e-02             0                 0.3  

business_id  open  review_count  stars  cat.Active Life  \
0  EmzaQR5hQlF0WIl24NxAZA     1            23    2.5                0   
1  KPoTixdjoJxSqRSEApSAGg     1           322    4.0                0   
2  e5kc0CQ4R-PCCDgb274gSg     1           191    3.5                0   
3  45puCRQ6Vh_IIAy7kkfFDQ     1           170    3.5                0   
4  BESmH-lmMm34_EefL8X9fQ     1           204    4.0                0   

   cat.Adult Entertainment  cat.Afghan  cat.African  cat.Airports  \
0                        0           0            0             0   
1                        0           0            0             0   
2                        0           0            0             0   
3                        0           0            0             0   
4                        0           0            0             0   

   cat.American (New)    ...      Noise.average  Noise.loud  Noise.quiet  \
0                   1    ...                  0           0            0   
1                   0    ...                  1           0            0   
2                   0    ...                  1           0            0   
3                   0    ...                  1           0            0   
4                   0    ...                  1           0            0   

   Noise.very_loud  beer_and_wine  full_bar  no alcohol  wifi.free  wifi.no  \
0                1              0         1           0          0        1   
1                0              0         1           0          0        1   
2                0              0         1           0          0        1   
3                0              0         1           0          0        1   
4                0              0         1           0          0        1   

   wifi.paid  
0          0  
1          0  
2          0  
3          0  
4          0  

[5 rows x 320 columns]

### Turn review topic_id_50 and topic_prob_50 columns into features

In [5]:
# topic_id_50 tells us which of the 50 topics the review is most about and 
# topic_prob_50 tells us the probability the review belongs to the topic
topic_cols = []
for index, row in reviews.loc[:,'topic_id_50':'topic_prob_50'].iterrows():
    d={}
    d[row[0]]=row[1]
    topic_cols.append(d)

topic_features = pd.DataFrame(topic_cols)
topic_features.fillna(0, inplace=True)
topic_features.columns = map(lambda x : 'topic_' + str(int(x)), topic_features.columns.values)
topic_features.head()

topic_0  topic_1  topic_2  topic_3  topic_4  topic_5  topic_6  topic_7  \
0        0        0        0        0        0      0.0        0        0   
1        0        0        0        0        0      0.0        0        0   
2        0        0        0        0        0      0.0        0        0   
3        0        0        0        0        0      0.0        0        0   
4        0        0        0        0        0      0.9        0        0   

   topic_8  topic_9    ...     topic_40  topic_41  topic_42  topic_43  \
0        0        0    ...            0         0         0       0.0   
1        0        0    ...            0         0         0       0.7   
2        0        0    ...            0         0         0       0.0   
3        0        0    ...            0         0         0       0.0   
4        0        0    ...            0         0         0       0.0   

   topic_44  topic_45  topic_46  topic_47  topic_48  topic_49  
0         0         0         0         0       0.0         0  
1         0         0         0         0       0.0         0  
2         0         0         0         0       0.0         0  
3         0         0         0         0       0.9         0  
4         0         0         0         0       0.0         0  

[5 rows x 50 columns]

In [27]:
# add topic features to our review data
reviews = pd.concat([reviews, topic_features], axis=1)
reviews

business_id               review_id date.year date.month  \
0     E7nk0lC317pGxMX56gvaVQ  onOLEBffcpWS_jcaBLCqdw      2008          6   
1     duHUQFn7K-Ybs1rPI4AOEQ  23z4mKkaMCXGFw4cm6xHxA      2012          4   
2     _K5cfCM7-zQZ1hXdbsc2fQ  dqrwXpcYdFvNYL5z1AeuZw      2013         12   
3     OTq55xUFMOz13DAi9D0_rw  uiJbTytfC1zsSOQ4aA3AWg      2014         12   
4     Kfbs3K37ZczrOQnoxgzNkw  II7cxU1cbOxVqzYY4ueflQ      2012          1   
5     eO-o7FRjNEzcSEG8pakkzA  B_82U0OC0N8a4hxlZPVYaQ      2014          7   
6     IcB2KDjGXdkG07Sitmik0A  fRBrsMowC6tedwrm5iwhhg      2013         11   
7     Exx5ffvnmk4MrTyCkPRuug  Ti-LQhvgGjoqN_i_sac8Yg      2013         11   
8     5_tsOGJQDDl9iKOr8K79Rw  g9LPuGXljBbFE213-qEQpA      2014         10   
9     EBwv1DBGKsfdNZEYL70Dqw  Oa1gS7cQ58y3J9Y4Fv524A      2011          6   
...                      ...                     ...       ...        ...   
9990  mwiNm868yAo8Xh8hO7Ke_Q  t6KcUB7tb6zeu0nKzWT6Rw      2010          2   
9991  lUnUMX5RveM7yvTGSyyJEQ  j7v2QDGPzkl8yYE3ulUCUw      2013          3   
9992  4dB7kQ7qtTkW2v7DX38r_w  TdDIm3UBaEsmxEVhzV5Bvg      2014          1   
9993  rrWhi2TL0smrbihvavZxcg  cdYWadDArNmwBVDH4kRcYA      2012          6   
9994  2WHP5nhS1rFszfRBKe6fWQ  ujpBMYH0mz3qQ2zuwcEndw      2014         12   
9995  yo33c2hiadxgZoYVQOpMPg  seF_ko-CXFF9Q8vO_-VabQ      2014          1   
9996  2mcmuAvpNIJuV3gV0v9v2A  uxPCpxVQs0tq-MIOuPANVg      2012         10   
9997  gUt-pPUpOVVhaCFC8-E4yQ  9Y7fo3mPm0o26Gd83YZvMQ      2013          9   
9998  Cr2JNnewX53TYd9w6qw6Jw  hDChdZ3C4Xz3dbzX5mA2UA      2013          1   
9999  QiNZaRZ5RmCfYiBgk3ApEQ  GZPPt-oQx91z5G9Pt5EIYw      2010         12   

     date.day  stars                                               text  \
0          17      1  This place is a huge waste of the space they m...   
1          15      3  I used to frequent Z-Tejas more frequently for...   
2           1      3  Meeting some friends for dinner, they live in ...   
3          23      4                        Good food, and good service   
4          25      4  Came here today to get lunch to go. They were ...   
5          17      5  I may be a tad biased, but in reality the beer...   
6          14      4  Came for lunch. We work right down the street ...   
7           2      5  Great place for a large group! Made reservatio...   
8          23      5  Awesome place! Amber was amazing! Gave suggest...   
9           2      2  Limited atmosphere, exceedingly slow, bland fo...   
...       ...    ...                                                ...   
9990       17      1  Not so good! I had the aligator and crawfish c...   
9991       23      3  Just okay mexican food. When visiting Arizona ...   
9992       31      4  Of all places for a country bar to open, Scott...   
9993       19      4  Been here a number of times and always blown a...   
9994        4      4  So good. Seriously. Came here last night for m...   
9995       17      1  Disappointing. . .we had high hopes for this p...   
9996        2      5  Best Vietnamese food in the west valley!  Pric...   
9997        7      4  Came here with my friends for a drink and were...   
9998       22      2  We are repeat customers. Today's visit was dis...   
9999        2      3  Great food. Service is excellent. Highly recom...   

      review_length  votes.cool  votes.funny    ...     topic_40 topic_41  \
0              1239           3            4    ...            0      0.0   
1               936           1            1    ...            0      0.0   
2              1119           0            0    ...            0      0.0   
3                27           0            0    ...            0      0.0   
4               661           1            0    ...            0      0.0   
5               412           0            0    ...            0      0.0   
6               358           0            0    ...            0      0.0   
7               637           0            0    ...    

In [7]:
for col in reviews.columns:
    print col
    print reviews[col].dtype

business_id
object
review_id
object
date.year
object
date.month
object
date.day
object
stars
object
text
object
review_length
object
votes.cool
object
votes.funny
object
votes.useful
object
user_id
object
topic_id_50
int64
topic_prob_50
float64
word_dist_50
object
topic_0
float64
topic_1
float64
topic_2
float64
topic_3
float64
topic_4
float64
topic_5
float64
topic_6
float64
topic_7
float64
topic_8
float64
topic_9
float64
topic_10
float64
topic_11
float64
topic_12
float64
topic_13
float64
topic_14
float64
topic_15
float64
topic_16
float64
topic_17
float64
topic_18
float64
topic_19
float64
topic_20
float64
topic_21
float64
topic_22
float64
topic_23
float64
topic_24
float64
topic_25
float64
topic_26
float64
topic_27
float64
topic_28
float64
topic_29
float64
topic_30
float64
topic_31
float64
topic_32
float64
topic_33
float64
topic_34
float64
topic_35
float64
topic_36
float64
topic_37
float64
topic_38
float64
topic_39
float64
topic_40
float64
topic_41
float64
topic_42
float64
topic_43
float

In [9]:
def change_to_int(df, col_names):
    for name in col_names:
        df[name] = df[name].astype(int)
    return

cols = ['stars', 'votes.cool', 'votes.funny', 'votes.useful', 'review_length']
change_to_int(reviews, cols)
reviews.dtypes

business_id      object
review_id        object
date.year        object
date.month       object
date.day         object
stars             int64
text             object
review_length     int64
...
topic_42    float64
topic_43    float64
topic_44    float64
topic_45    float64
topic_46    float64
topic_47    float64
topic_48    float64
topic_49    float64
Length: 65, dtype: object

## Join the topic distribution to business data
#### Note: At this point, we have 3,358 restaurants and 10,000 reviews from which I will build my models.

In [10]:
grouped_reviews = reviews.groupby('business_id').agg([np.sum, np.mean])
grouped_reviews.reset_index(inplace=True)

In [11]:
grouped_reviews.head()

business_id stars      review_length        votes.cool       \
                            sum mean           sum   mean        sum mean   
0  --5jkZ3-nUPZxUvtcbr8Uw     5  5.0            44   44.0          0  0.0   
1  -1bOb2izeJBZjHC7NWxiPA     3  3.0           393  393.0          0  0.0   
2  -2gyHPSL8pJjkSe5Ozzw6Q    11  3.7          1088  362.7          0  0.0   
3  -3WVw1TNQbPBzaKCaQQ1AQ    36  4.0          5046  560.7          3  0.3   
4  -3frEL8Vo8kyq4kfgTn0GA     3  3.0           315  315.0          3  3.0   

  votes.funny      votes.useful ...  topic_45      topic_46      topic_47  \
          sum mean          sum ...       sum mean      sum mean      sum   
0           0  0.0            0 ...         0    0      0.0  0.0        0   
1           0  0.0            0 ...         0    0      0.0  0.0        0   
2           0  0.0            2 ...         0    0      0.0  0.0        0   
3           4  0.4            5 ...         0    0      0.6  0.1        0   
4           1  1.0            2 ...         0    0      0.0  0.0        0   

       topic_48      topic_49       
  mean      sum mean      sum mean  
0    0      0.0  0.0        0    0  
1    0      0.0  0.0        0    0  
2    0      0.3  0.1        0    0  
3    0      0.0  0.0        0    0  
4    0      0.0  0.0        0    0  

[5 rows x 115 columns]

In [19]:
grouped_reviews.columns

MultiIndex(levels=[[u'stars', u'review_length', u'votes.cool', u'votes.funny', u'votes.useful', u'topic_id_50', u'topic_prob_50', u'topic_0', u'topic_1', u'topic_2', u'topic_3', u'topic_4', u'topic_5', u'topic_6', u'topic_7', u'topic_8', u'topic_9', u'topic_10', u'topic_11', u'topic_12', u'topic_13', u'topic_14', u'topic_15', u'topic_16', u'topic_17', u'topic_18', u'topic_19', u'topic_20', u'topic_21', u'topic_22', u'topic_23', u'topic_24', u'topic_25', u'topic_26', u'topic_27', u'topic_28', u'topic_29', u'topic_30', u'topic_31', u'topic_32', u'topic_33', u'topic_34', u'topic_35', u'topic_36', u'topic_37', u'topic_38', u'topic_39', u'topic_40', u'topic_41', u'topic_42', u'topic_43', u'topic_44', u'topic_45', u'topic_46', u'topic_47', u'topic_48', u'topic_49', u'business_id'], [u'sum', u'mean', u'']],
           labels=[[57, 0, 0, 1, 1, 2, 2, 3, 3, 4, 4, 5, 5, 6, 6, 7, 7, 8, 8, 9, 9, 10, 10, 11, 11, 12, 12, 13, 13, 14, 14, 15, 15, 16, 16, 17, 17, 18, 18, 19, 19, 20, 20, 21, 21, 22, 22, 

In [13]:
review_means = grouped_reviews.xs('mean', level=1, axis=1)
review_means['business_id'] = grouped_reviews['business_id'].values
review_means

/Users/aprilsong/anaconda/lib/python2.7/site-packages/IPython/kernel/__main__.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  from IPython.kernel.zmq import kernelapp as app


stars  review_length  votes.cool  votes.funny  votes.useful  \
0       5.0           44.0         0.0          0.0           0.0   
1       3.0          393.0         0.0          0.0           0.0   
2       3.7          362.7         0.0          0.0           0.7   
3       4.0          560.7         0.3          0.4           0.6   
4       3.0          315.0         3.0          1.0           2.0   
5       3.7         1195.0         0.0          0.0           0.0   
6       4.5          151.5         0.5          0.0           0.5   
7       2.0         1643.0         0.0          0.0           0.0   
8       5.0          440.0         1.0          1.0           1.0   
9       5.0          123.0         0.0          1.0           1.0   
...     ...            ...         ...          ...           ...   
3348    4.3          301.3         0.0          0.0           0.0   
3349    2.8          526.8         0.0          0.0           0.2   
3350    3.7          280.9         0.0          0.0           0.4   
3351    4.6          509.5         0.0          0.0           0.2   
3352    5.0          483.0         0.0          0.0           0.0   
3353    4.5          426.5         0.5          0.5           0.5   
3354    4.0          508.0         0.0          0.0           0.0   
3355    5.0          276.0         0.0          0.0           0.0   
3356    3.5          271.0         0.5          0.0           0.0   
3357    4.5          267.0         0.0          0.0           0.0   

      topic_id_50  topic_prob_50  topic_0  topic_1  topic_2  \
0            41.0            0.6        0        0      0.0   
1            44.0            0.5        0        0      0.0   
2            37.0            0.4        0        0      0.0   
3            28.1            0.6        0        0      0.0   
4            18.0            0.2        0        0      0.0   
5            42.7            0.4        0        0      0.0   
6            40.5            0.9        0        0      0.0   
7            37.0            0.4        0        0      0.0   
8            33.0            0.6        0        0      0.0   
9            35.0            0.2        0        0      0.0   
...           ...            ...      ...      ...      ...   
3348         41.7            0.7        0        0      0.0   
3349         18.5            0.4        0        0      0.0   
3350         29.7            0.5        0        0      0.0   
3351         20.4            0.6        0        0      0.0   
3352          4.0            0.6        0        0      0.3   
3353         32.0            0.5        0        0      0.0   
3354         31.5            0.8        0        0      0.0   
3355         39.0            0.8        0        0      0.0   
3356         23.2            0.6        0        0      0.2   
3357         33.0            0.5        0        0      0.0   

               ...            topic_41  topic_42  topic_43  topic_44  \
0              ...                 0.6         0       0.0       0.0   
1              ...                 0.0         0       0.0       0.5   
2              ...                 0.0         0       0.0       0.0   
3              ...                 0.0         0       0.0       0.0   
4              ...                 0.0         0       0.0       0.0   
5              ...                 0.0         0       0.0       0.0   
6              ...                 0.5         0       0.0       0.0   
7              ...                 0.0         0       0.0       0.0   
8              ...                 0.0         0       0.0       0.0   
9              ...                 0.0         0       0.0       0.0   
...            ...                 ...       ...       ...       ...   
3348           ...                 0.2         0       0.0       0.0   
3349           ...                 0.0         0       0.0       0.0   
3350           ...                 0.1         0       0.0       0.0   
3351           ...  

In [14]:
review_means.columns = map(lambda x : 'avg_' + x, review_means.columns.values)
review_means

avg_stars  avg_review_length  avg_votes.cool  avg_votes.funny  \
0           5.0               44.0             0.0              0.0   
1           3.0              393.0             0.0              0.0   
2           3.7              362.7             0.0              0.0   
3           4.0              560.7             0.3              0.4   
4           3.0              315.0             3.0              1.0   
5           3.7             1195.0             0.0              0.0   
6           4.5              151.5             0.5              0.0   
7           2.0             1643.0             0.0              0.0   
8           5.0              440.0             1.0              1.0   
9           5.0              123.0             0.0              1.0   
...         ...                ...             ...              ...   
3348        4.3              301.3             0.0              0.0   
3349        2.8              526.8             0.0              0.0   
3350        3.7              280.9             0.0              0.0   
3351        4.6              509.5             0.0              0.0   
3352        5.0              483.0             0.0              0.0   
3353        4.5              426.5             0.5              0.5   
3354        4.0              508.0             0.0              0.0   
3355        5.0              276.0             0.0              0.0   
3356        3.5              271.0             0.5              0.0   
3357        4.5              267.0             0.0              0.0   

      avg_votes.useful  avg_topic_id_50  avg_topic_prob_50  avg_topic_0  \
0                  0.0             41.0                0.6            0   
1                  0.0             44.0                0.5            0   
2                  0.7             37.0                0.4            0   
3                  0.6             28.1                0.6            0   
4                  2.0             18.0                0.2            0   
5                  0.0             42.7                0.4            0   
6                  0.5             40.5                0.9            0   
7                  0.0             37.0                0.4            0   
8                  1.0             33.0                0.6            0   
9                  1.0             35.0                0.2            0   
...                ...              ...                ...          ...   
3348               0.0             41.7                0.7            0   
3349               0.2             18.5                0.4            0   
3350               0.4             29.7                0.5            0   
3351               0.2             20.4                0.6            0   
3352               0.0              4.0                0.6            0   
3353               0.5             32.0                0.5            0   
3354               0.0             31.5                0.8            0   
3355               0.0             39.0                0.8            0   
3356               0.0             23.2                0.6            0   
3357               0.0             33.0                0.5            0   

      avg_topic_1  avg_topic_2           ...            avg_topic_41  \
0               0          0.0           ...                     0.6   
1               0          0.0           ...                     0.0   
2               0          0.0           ...                     0.0   
3               0          0.0           ...                     0.0   
4               0          0.0           ...                     0.0   
5               0          0.0           ...                     0.0   
6               0          0.0           ...                     0.5   
7               0          0.0           ...                     0.0   
8               0          0.0           ...                     0.0   
9               0          0.0           ...                     0.0   


In [15]:
restaurants

business_id  open  review_count  stars  cat.Active Life  \
0     EmzaQR5hQlF0WIl24NxAZA     1            23    2.5                0   
1     KPoTixdjoJxSqRSEApSAGg     1           322    4.0                0   
2     e5kc0CQ4R-PCCDgb274gSg     1           191    3.5                0   
3     45puCRQ6Vh_IIAy7kkfFDQ     1           170    3.5                0   
4     BESmH-lmMm34_EefL8X9fQ     1           204    4.0                0   
5     D09yG6Z3gcsh24Qn7Y4gYA     1            73    3.0                0   
6     kaIue7GRCmkPzDeHDBTttQ     1           152    4.0                0   
7     j8nyAaN_w5t6wYv-ni23sg     0            77    3.0                0   
8     IceZ4BtTz76eppjNWm-c6A     0            59    3.5                0   
9     -lOSaCuBRAvX5JBifx-EMw     1           105    3.0                0   
...                      ...   ...           ...    ...              ...   
3348  WlPYm5aK-ewwWeO4TlrF1g     1             7    5.0                0   
3349  mMJVKgRNlsr0YrkUBGqfQw     1            37    3.5                0   
3350  d7b78XiwyCApbKbWkuCGuw     1            36    3.0                0   
3351  ETrtpthOGuC8q7rE17s30g     1            55    3.5                0   
3352  u5cRdcNN-9V8SlsfO_Kl7w     1            40    4.5                0   
3353  n9H3Ay_jMjhKjUNZWtaICA     1            19    4.0                0   
3354  hQBLiFMHKq7OvsaGCb9AZA     1            69    4.0                0   
3355  tc1kER4tWn3LI99Rk-NVZw     1            37    4.0                0   
3356  q4MTywvT1EFuWiFMLQ7PXA     1            19    4.5                0   
3357  qCPBS-m_4uDO0EgIYGtoxw     1            28    4.0                1   

      cat.Adult Entertainment  cat.Afghan  cat.African  cat.Airports  \
0                           0           0            0             0   
1                           0           0            0             0   
2                           0           0            0             0   
3                           0           0            0             0   
4                           0           0            0             0   
5                           0           0            0             0   
6                           0           0            0             0   
7                           0           0            0             0   
8                           0           0            0             0   
9                           0           0            0             0   
...                       ...         ...          ...           ...   
3348                        0           0            0             0   
3349                        0           0            0             0   
3350                        0           0            0             0   
3351                        0           0            0             0   
3352                        0           0            0             0   
3353                        0           0            0             0   
3354                        0           0            0             0   
3355                        0           0            0             0   
3356                        0           0            0             0   
3357                        0           0            0             0   

      cat.American (New)    ...      Noise.average  Noise.loud  Noise.quiet  \
0                      1    ...                  0           0            0   
1                      0    ...                  1           0            0   
2                      0    ...                  1           0            0   
3                      0    ...                  1           0            0   
4                      0    ...                  1           0            0   
5                      0    ...                  0           0            1   
6                      0    ...                  1           0            0   
7                      1    ...                  1           0            0   
8                      0    ...                 

In [17]:
restaurants = pd.merge(left=restaurants, right=review_means, how='inner', left_on='business_id', right_on='avg_business_id')
restaurants.to_pickle('restaurant/restaurant_with_topic_dist1.pkl')
restaurants


business_id  open  review_count  stars  cat.Active Life  \
0     EmzaQR5hQlF0WIl24NxAZA     1            23    2.5                0   
1     KPoTixdjoJxSqRSEApSAGg     1           322    4.0                0   
2     e5kc0CQ4R-PCCDgb274gSg     1           191    3.5                0   
3     45puCRQ6Vh_IIAy7kkfFDQ     1           170    3.5                0   
4     BESmH-lmMm34_EefL8X9fQ     1           204    4.0                0   
5     D09yG6Z3gcsh24Qn7Y4gYA     1            73    3.0                0   
6     kaIue7GRCmkPzDeHDBTttQ     1           152    4.0                0   
7     j8nyAaN_w5t6wYv-ni23sg     0            77    3.0                0   
8     IceZ4BtTz76eppjNWm-c6A     0            59    3.5                0   
9     -lOSaCuBRAvX5JBifx-EMw     1           105    3.0                0   
...                      ...   ...           ...    ...              ...   
3348  WlPYm5aK-ewwWeO4TlrF1g     1             7    5.0                0   
3349  mMJVKgRNlsr0YrkUBGqfQw     1            37    3.5                0   
3350  d7b78XiwyCApbKbWkuCGuw     1            36    3.0                0   
3351  ETrtpthOGuC8q7rE17s30g     1            55    3.5                0   
3352  u5cRdcNN-9V8SlsfO_Kl7w     1            40    4.5                0   
3353  n9H3Ay_jMjhKjUNZWtaICA     1            19    4.0                0   
3354  hQBLiFMHKq7OvsaGCb9AZA     1            69    4.0                0   
3355  tc1kER4tWn3LI99Rk-NVZw     1            37    4.0                0   
3356  q4MTywvT1EFuWiFMLQ7PXA     1            19    4.5                0   
3357  qCPBS-m_4uDO0EgIYGtoxw     1            28    4.0                1   

      cat.Adult Entertainment  cat.Afghan  cat.African  cat.Airports  \
0                           0           0            0             0   
1                           0           0            0             0   
2                           0           0            0             0   
3                           0           0            0             0   
4                           0           0            0             0   
5                           0           0            0             0   
6                           0           0            0             0   
7                           0           0            0             0   
8                           0           0            0             0   
9                           0           0            0             0   
...                       ...         ...          ...           ...   
3348                        0           0            0             0   
3349                        0           0            0             0   
3350                        0           0            0             0   
3351                        0           0            0             0   
3352                        0           0            0             0   
3353                        0           0            0             0   
3354                        0           0            0             0   
3355                        0           0            0             0   
3356                        0           0            0             0   
3357                        0           0            0             0   

      cat.American (New)           ...            avg_topic_41  avg_topic_42  \
0                      1           ...                     0.0             0   
1                      0           ...                     0.0             0   
2                      0           ...                     0.0             0   
3                      0           ...                     0.0             0   
4                      0           ...                     0.1             0   
5                      0           ...                     0.0             0   
6                      0           ...                     0.0             0   
7                      1           ...                     0.0             0   
8                      0           ... 

### Create initial feature matrix

In [176]:
print restaurants.columns.values

[u'business_id' u'open' u'review_count' u'stars' u'cat.Active Life'
 u'cat.Adult Entertainment' u'cat.Afghan' u'cat.African' u'cat.Airports'
 u'cat.American (New)' u'cat.American (Traditional)' u'cat.Amusement Parks'
 u'cat.Appliances' u'cat.Arabian' u'cat.Arcades' u'cat.Argentine'
 u'cat.Art Galleries' u'cat.Arts & Crafts' u'cat.Arts & Entertainment'
 u'cat.Asian Fusion' u'cat.Australian' u'cat.Auto Repair' u'cat.Automotive'
 u'cat.Bagels' u'cat.Bakeries' u'cat.Bangladeshi' u'cat.Barbeque'
 u'cat.Bars' u'cat.Basque' u'cat.Beauty & Spas' u'cat.Bed & Breakfast'
 u'cat.Beer Bar' u'cat.Beer Gardens' u'cat.Beer, Wine & Spirits'
 u'cat.Belgian' u'cat.Bistros' u'cat.Bowling' u'cat.Brasseries'
 u'cat.Brazilian' u'cat.Breakfast & Brunch' u'cat.Breweries' u'cat.British'
 u'cat.Bubble Tea' u'cat.Buffets' u'cat.Burgers' u'cat.Burmese'
 u'cat.Butcher' u'cat.Cafes' u'cat.Cafeteria' u'cat.Cajun/Creole'
 u'cat.Cambodian' u'cat.Candy Stores' u'cat.Cantonese' u'cat.Car Wash'
 u'cat.Caribbean' u'cat.Cas

In [177]:
del_cols = ['business_id', 'avg_stars', 'avg_topic_id_50', 'avg_topic_prob_50', 'avg_business_id']
for col in del_cols:
    del restaurants[col]


In [178]:
# check that our data types are all floats/integers
print restaurants.dtypes.values


[dtype('int64') dtype('int64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dtype('float64') dtype('float64')
 dtype('float64') dtype('float64') dty

In [179]:
restaurants.to_pickle('restaurant/restaurant_with_topic_dist.pkl')